packages/components/src/__alpha__/SingleSelect/subcomponents/ListSection/ListSection.tsx

```diff
@@ -1,25 +1,47 @@
-import React, { type PropsWithChildren } from 'react'
-import classNames from 'classnames'
-import {
-  Header as RACHeader,
-  ListBoxSection as RACListBoxSection,
-  type ListBoxSectionProps,
-} from 'react-aria-components'
-import { type SelectSection } from '../../types'
+import React from 'react'
+import { useListBoxSection } from 'react-aria'
+import { type ListSectionProps, type SelectItem } from '../../types'
+import { ListItem } from '../ListItem'
 import styles from './ListSection.module.css'
 
-export const ListSection = ({
-  name,
-  className,
-  children,
-  ...props
-}: ListBoxSectionProps<SelectSection> &
-  PropsWithChildren & { name: string }): React.ReactElement => {
+export const ListSection = <T extends SelectItem>({
+  section,
+  state,
+}: ListSectionProps<T>): JSX.Element => {
+  const { headingProps, itemProps, groupProps } = useListBoxSection({
+    'heading': section.rendered,
+    'aria-label': section['aria-label'],
+  })
+
+  // Known issue: Incorrect group count screenreader announcements with all grouped ListBox patterns in Chrome & Safari
+  // See https://github.com/adobe/react-spectrum/issues/1234 for more details
+
   return (
-    <RACListBoxSection {...props}>
-      <RACHeader className={classNames(styles.listSectionHeader, className)}>{name}</RACHeader>
-      {children}
-    </RACListBoxSection>
+    <>
+      {section.key !== state.collection.getFirstKey() && <li role="presentation" aria-hidden />}
+      <li {...itemProps}>
+        {section.rendered && (
+          <span
+            {...headingProps}
+            role="presentation"
+            aria-hidden
+            className={styles.listSectionHeader}
+          >
+            {section.rendered}
+          </span>
+        )}
+        <ul
+          {...groupProps}
+          style={{
```

Comment:
Can we change the inline styles to CSS modules. I know we're not quite at full styling stage but just don't won't to miss this in future reviews 😉